### DISCUSSION TITLE
Error while installing LDT (compiling)

### DISCUSSION THREAD

**Discussion by MariRiba** [NASA-LIS/LISF#1379] [General]
Hi , I am trying to install LISF (downloaded from "git clone https://github.com/NASA-LIS/LISF") master branch.
I have installed all the dependencies and successfully configured LDT.
But I am getting this error while compiling it.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Compiling LDT

[INFO] Compiling LDT source code

gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_mpiMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_logMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_timeMgrMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_coreMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_DAobsDataMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/  -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/  -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/  -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/  -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/  -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../obsSim/OSSEmask/AMSR2/AMSR2OSSEmask_Mod.F90
../obsSim/OSSEmask/AMSR2/AMSR2OSSEmask_Mod.F90:24:7:

   24 |   use LDT_constantsMod, only : LDT_CONST_PATH_LEN
      |       1
Fatal Error: Cannot open module file ‘ldt_constantsmod.mod’ for reading at (1): No such file or directory
compilation terminated.
make: *** [Makefile:108: AMSR2OSSEmask_Mod.o] Error 1

[ERR] Compile failed
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
My configure

(vvv) root@mari-VivoBook-ASUSLaptop-X509DA-M509DA:/home/mari/Desktop/project/LISF/ldt# ./configure
------------------------------------------------------------------------
Setting up configuration for LDT
Parallelism (0-serial, 1-dmpar, default=0): 
Optimization level (-3=strict checks with warnings, -2=strict checks, -1=debug, 0,1,2,3, default=2): 
Assume little/big_endian data format (1-little, 2-big, default=2): 
Use GRIBAPI/ECCODES? (0-neither, 1-gribapi, 2-eccodes, default=2): 
NETCDF version (3 or 4, default=4)?: 
NETCDF use shuffle filter? (1-yes, 0-no, default = 1): 
NETCDF use deflate filter? (1-yes, 0-no, default = 1): 
NETCDF use deflate level? (1 to 9-yes, 0-no, default = 9): 
Use HDF4? (1-yes, 0-no, default=1): 
Use HDF5? (1-yes, 0-no, default=1): 
Use HDFEOS? (1-yes, 0-no, default=1): 
Enable GeoTIFF support? (1-yes, 0-no, default=1): 
Enable LIBGEOTIFF support? (1-yes, 0-no, default=1): 
Include date/time stamp history? (1-yes, 0-no, default=1): 
-----------------------------------------------------
 configure.ldt file generated successfully
-----------------------------------------------------
Settings are written to configure.ldt in the make directory.
If you wish to change settings, please edit that file.

To compile, run the compile script.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

My env variables path are as follows:

gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0

export LDT_ARCH=linux_gfortran
export LDT_FC=gcc
export LDT_CC=gcc
export LDT_MODESMF=/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
export LDT_LIBESMF=/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
export LDT_OPENJPEG=/home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/src/
export LDT_ECCODES=/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/
export LDT_NETCDF=/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/
export LDT_HDF4=/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/
export LDT_HDF5=/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/
export LDT_GDAL=/home/mari/Desktop/project/Dependencies/
export LDT_HDFEOS=/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0/
export LDT_FORTRANGIS=/home/mari/Desktop/project/Dependencies/fortrangis-2.5/
export LDT_LIBGEOTIFF=/home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/
export LDT_JPEG=/usr/lib/x86_64-linux-gnu/

-----------------------------------------------------------------------------------------------------------------------
 And my configure.ldt

FC              = gcc
FC77            = gcc
LD              = gcc
CC              = gcc
AR              = ar
MOD_ESMF        = /home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
LIB_ESMF        = /home/mari/Desktop/project/Dependencies/esmf/lib/libO/Linux.gfortran.64.mpiuni.default/
INC_JPEG2000      = /home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/include/
LIB_JPEG2000      = /home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/lib/
INC_ECCODES     = /home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/include/
LIB_ECCODES     = /home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/lib/
INC_NETCDF      = /home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/include/
LIB_NETCDF      = /home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/lib/
INC_HDF4        = /home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/include/
LIB_HDF4        = /home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/lib/
INC_HDF5        = /home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/include/
LIB_HDF5        = /home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/lib/
INC_HDFEOS      = /home/mari/Desktop/project/Dependencies/hdf-eos5-2.0/include/
LIB_HDFEOS      = /home/mari/Desktop/project/Dependencies/hdf-eos5-2.0/lib/
INC_FORTRANGIS1 = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/include/
INC_FORTRANGIS2 = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/include/
LIB_FORTRANGIS  = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/lib/
INC_LIBGEOTIFF  = /home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/include/
LIB_LIBGEOTIFF  = /home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/lib/
LIB_GDAL        = /home/mari/Desktop/project/Dependencies/gdal-3.7.0/lib/
CFLAGS          = -c -DGFORTRAN  -I$(INC_LIBGEOTIFF)
FFLAGS77        = -c -pass-exit-codes -O2   -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF)  -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
FFLAGS          =  -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF)  -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
LDFLAGS         =  -L$(LIB_ESMF) -lesmf -lstdc++ -lz  -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -lopenjp2 -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz  -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl -L$(LIB_FORTRANGIS) -lfortrangis -lfortranc -L$(LIB_GDAL) -lgdal -L$(LIB_LIBGEOTIFF)  -ltiff -lgeotiff -lm -lz -ljpeg 
-------------------------------------------------------------------------------------------------------------------------------------------------------------
Please Help !!!!!!!!

**Comment by sudarsan3010** [comment]
I would recommend to first try to compile only with the necessary libraries like ESMF, Openjpeg, netcdf and eccodes, once it got compiled, now you can also add the extra libraries needed for your work and compile it again.

I have recently installed LISF 7.4.2 in my HPC and all executables has been created, but facing issues in running those executables. I could share my env variables and library versions installed which might be useful for you.

GNU compiler collection version:
GNU Fortran (gfortran) - 9.3.0
GCC (gcc; mpicxx) - 9.3.0


Netcdf-c 4.7.4
Netcdf-fortran 4.5.3
Openjpeg 2.4.0
Eccodes 2.19
ESMF 8.2.0
LISF 7.4.2

export LDT_ARCH=linux_gfortran
export LDT_FC=gfortran
export LDT_CC=gcc

**Reply by MariRiba** [reply]
I have tried compiling with only the required libraries but i am still getting the same error.

Setting up configuration for LDT
Parallelism (0-serial, 1-dmpar, default=0):
Optimization level (-3=strict checks with warnings, -2=strict checks, -1=debug, 0,1,2,3, default=2):
Assume little/big_endian data format (1-little, 2-big, default=2):
Use GRIBAPI/ECCODES? (0-neither, 1-gribapi, 2-eccodes, default=2):
NETCDF version (3 or 4, default=4)?:
NETCDF use shuffle filter? (1-yes, 0-no, default = 1):
NETCDF use deflate filter? (1-yes, 0-no, default = 1):
NETCDF use deflate level? (1 to 9-yes, 0-no, default = 9):
Include date/time stamp history? (1-yes, 0-no, default=1):

The rest options I choose (0-no) , but I am still getting the same error.
